### PR TITLE
Removed console.log() from transport doc

### LINKED ISSUE #1200: Missing log output when using custom transports
Errors don't end up in `stdout` when using pino-pretty or a custom transport after catching en error (e.g. typical express + mongoose app setup)

The following is the `README.md` from [this issue's repro repository](https://github.com/replete/pino-missing-log-output), which explains the problem:

<hr/>

# Missing log output issue

This repo has three express applications, each identical apart from the configuration passed into `pino()`.

The problem is that when using a custom transport, in this scenario, no output is logged.

```js
app.listen(3000, async () => {
	log.info("API listening...");
	await connectDatabase();
});

async function connectDatabase() {
	try {
		// Imagine this is mongoose.connect(), and its thrown an
		// Authentication Failed error, but no errror is logged. 
		throw Error('some error')
		log.info("Connected to Database...");
	} catch (error) {
		log.error(error); // never logged with custom pino transports... 
		process.exit(1);
	}
}
```

### Install
``` bash
git clone git@github.com:replete/pino-missing-log-output.git
cd pino-missing-log-output
npm install
```

### Run
```bash
npm run appWithPino
npm run appWithPinoPretty
npm run appWithPinoCustom
```

## Pino with default configuration:

```js
const log = pino();
```

Running `npm run appWithPino` results in expected behaviour.

```bash
$ npm run appWithPino

> pino-problem@1.0.0 appWithPino
> node appWithPino.js

{"level":30,"time":1635621351990,"pid":83045,"hostname":"Seneca","msg":"App start"}
{"level":30,"time":1635621351996,"pid":83045,"hostname":"Seneca","msg":"API listening..."}
{"level":50,"time":1635621351996,"pid":83045,"hostname":"Seneca","err":{"type":"Error","message":"some error","stack":"Error: some error\n    at connectDatabase (/Users/phil/dev/pino-problem/appWithPino.js:17:9)\n    at Server.<anonymous> (/Users/phil/dev/pino-problem/appWithPino.js:12:8)\n    at Object.onceWrapper (node:events:509:28)\n    at Server.emit (node:events:390:28)\n    at emitListeningNT (node:net:1368:10)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)"},"msg":"some error"}
```

## Pino configured with pretty-print:

```js
const log = pino({
	transport: {
		target: "pino-pretty",
		options: { destination: 1, colorize: true, ignore: "pid,hostname" },
	},
});
```

Running `npm run appWithPinoPretty` results in missing logs:

```bash
$ npm run appWithPinoCustom

> pino-problem@1.0.0 appWithPinoCustom
> node appWithPinoCustom.js
```

## Pino configured with custom transport:

```js
const log = pino({
	transport: {
		pipeline: [
			{
				// logger output for terminal, colorised by level for easy reading
				target: "./pinocustom.mjs",
				options: {
					destination: 1,
				},
			},
		],
	},
});
```

Running `npm run appWithPinoCustom` results in even more missing logs:

```bash
$ npm run appWithPinoCustom

> pino-problem@1.0.0 appWithPinoCustom
> node appWithPinoCustom.js

```

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #1200

### CODE REVIEW

File: docs/transports.md [hunk 1]
```diff
@@ -116,11 +111,27 @@ You can see an example using a async iterator with ESM:
 
 ```js
 import build from 'pino-abstract-transport'
+import SonicBoom from 'sonic-boom'
+import { once } from 'events'
 
 export default async function (opts) {
+  // SonicBoom is necessary to avoid loops with the main thread.
+  // It is the same of pino.destination().
+  const destination = new SonicBoom({ dest: opts.destination || 1, sync: false })
```
Comment (jsumners) [inline docs/transports.md:120]: Why don't we show using `pino.destination` here instead?

Comment (mcollina) [reply docs/transports.md:120]: If I am writing a module to be published on npm I do not want to depend on pino. This should probably be written somewhere.

Comment (jsumners) [reply docs/transports.md:120]: That makes sense. I think these docs as written, though, are from the perspective of someone writing an _application_ that depends on Pino. Thus, they will already have the dependency.

Comment (mcollina) [reply docs/transports.md:120]: Should we mention both?

Comment (jsumners) [reply docs/transports.md:120]: Yeah, but we can follow up with another PR at a later date if you like.

Comment (mcollina) [reply docs/transports.md:120]: This page needs an overhaul I think :(. Let's land this and iterate.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Pull request merged**